packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,249 @@
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+  position: relative;
+  pointer-events: all;
+`;
+
+const modalSizes: { readonly [K in ModalSize]: string } = {
+  small: css`
+    width: 400px;
+  `,
+
+  default: css`
+    width: 720px;
+  `,
+
+  large: css`
+    ${mq({
+      width: ['720px', '720px', '960px'],
+    })}
+  `,
+};
+
+const closeButton = css`
+  color: ${uiColors.gray.dark1};
+  position: absolute;
+  cursor: pointer;
+  right: 12px;
+  top: 12px;
+`;
+
+const escapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  size="large"
+  setActive={setActive}
+  shouldClose={() => console.log('Modal is closing now!')}
+  >
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.size String to determine size of Modal. ['small', 'default', 'large']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.shouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  size = ModalSize.Default,
+  setActive,
+  children,
+  shouldClose,
+  className,
+  ...rest
+}: ModalProps) {
+  if (!active) {
+    return null;
+  }
+
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    // Don't close modal if shouldClose returns false or no setActive callback is passed
+    if (!setActive || (shouldClose && !shouldClose())) {
+      return;
+    }
+
+    setActive(false);
+  }, [setActive, shouldClose]);
+
+  const handleBackdropClick = (e: React.SyntheticEvent) => {
+    if (!scrollContainerRef.current) {
+      return;
+    }
+
+    if (e.target === scrollContainerRef.current) {
+      handleClose();
+    }
+  };
+
+  const handleEscape = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.keyCode === escapeKey) {
+        handleClose();
+      }
+    },
+    [handleClose],
+  );
+
+  useEventListener('keydown', handleEscape, {
+    options: { once: true },
```

Comment:
Do we want once here? What if a user types in 'f'? Won't that remove this event listener?

Reply:
You're correct. Removed!